src/test/java/school/cesar/eta/unit/PersonTest.java

```diff
@@ -1,62 +1,132 @@
 package school.cesar.eta.unit;
 
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Random;
+
 import static org.junit.jupiter.api.Assertions.fail;
 
 public class PersonTest {
+    Person a;
+    @BeforeEach
+    public void setup(){
+        a = new Person();
+    }
+
     @Test
     public void getName_firstNameJonLastNameSnow_jonSnow() {
-        fail();
+        a.setName("Jon");
+        a.setLastName("Snow");
+
+        Assertions.assertEquals("Jon Snow" , a.getName());
     }
 
     @Test
     public void getName_firstNameJonNoLastName_jon() {
-        fail();
+        a.setName("Jon");
+
+        Assertions.assertEquals("Jon" , a.getName());
     }
 
     @Test
     public void getName_noFirstNameLastNameSnow_snow() {
-        fail();
+        a.setLastName("Snow");
+
+        Assertions.assertEquals("Snow", a.getName());
     }
 
     @Test
     public void getName_noFirstNameNorLastName_throwsException() {
-        fail();
-    }
 
-    @Test
-    public void isBirthdayToday_differentMonthAndDay_false() {
-        fail();
+        Assertions.assertThrows(RuntimeException.class,() -> new Person().getName() );
     }
 
+    //@Test
+    //public void isBirthdayToday_differentMonthAndDay_false() { fail(); }
+
     @Test
     public void isBirthdayToday_sameMonthDifferentDay_false() {
-        fail();
+        ArrayList<String> listNum = new ArrayList<String>();
+        for (int x = 1; x <= 9; x++) {
+            listNum.add(String.valueOf(x));
+        }
+
+        listNum.remove(String.valueOf(LocalDate.now().getDayOfMonth()));
+        Random y = new Random();
+        int day = y.nextInt(7);
+
+        //LocalDate dataManipulacao = LocalDate.now();
+        //dataManipulacao.plusDays(5).plusMonths(1);
+
+        String date = LocalDate.now().getYear() + "-" + LocalDate.now().getMonthValue() + "-0" +  listNum.get(day);
+        a.setBirthday(LocalDate.parse(date));
+
+        Assertions.assertFalse(a.isBirthdayToday());
     }
 
     @Test
     public void isBirthdayToday_sameMonthAndDay_true() {
-        fail();
+        a.setBirthday(LocalDate.now());
+
+        Assertions.assertTrue(a.isBirthdayToday());
     }
 
     @Test
     public void addToFamily_somePerson_familyHasNewMember() {
-        fail();
+        final boolean[] newMember = {false};
+        a = new Person(){
+            @Override
+            public void addToFamily(Person a1) {
+                newMember[0] = true;
+                super.addToFamily(a1);
+            }
+        };
+        a.addToFamily(new Person());
+
+        Assertions.assertTrue(newMember[0]);
```

Comment:
Sempre que o método `addToFamily()` for chamado, `newMember[0]`  retornará `true`. O que não significa que o método `addToFamily()` foi corretamente implementado.

**Verifique se o atributo `family` contém a pessoa adicionada**